src/python/gudhi/cover_complex.py

```diff
@@ -17,6 +17,10 @@
 from sklearn.metrics         import pairwise_distances
 from scipy.spatial.distance  import directed_hausdorff
 
+from sklearn import __version__ as sklearn_version
+from pkg_resources import packaging
+agglomerative_clustering_metric = packaging.version.parse(sklearn_version) >= packaging.version.parse('1.2.0')
```

Comment:
@mglisse you are right and pep 386 is superseded... Sorry @MathieuCarriere for the bad pointer... sklearn has its own `parse_version` implementation that seems to be the right one to use:
```suggestion
from sklearn.utils.fixes import parse_version
agglomerative_clustering_metric = parse_version(sklearn_version) >= parse_version('1.2.0')
```

Reply:
No worries! I've just done the change.